app/src/main/java/com/quran/labs/androidquran/service/QuranDownloadService.java

```diff
@@ -233,6 +231,12 @@ private void sendNoOpMessage(int id) {
 
   @Override
   public int onStartCommand(Intent intent, int flags, int startId) {
+    // if it's a download, it wants to be a foreground service.
+    // quickly start as foreground before actually enqueueing the request.
+    if (ACTION_DOWNLOAD_URL.equals(intent.getAction())) {
```

Comment:
this is actually a lot sooner than where it was happening before - before, `handleOnStartCommand` would process the message and then give it to the handler, which would eventually start foreground and show the notification. on some devices, this could take so long, so make it the first thing we do.